aspnetcore/tutorials/web-api-vsc.md

```diff
@@ -0,0 +1,272 @@
+
+[!code-csharp[Main](first-web-api/sample/TodoApi/Models/ITodoRepository.cs)]
+
+This interface defines basic CRUD operations.
+
+Add a `TodoRepository` class that implements `ITodoRepository`:
+
+[!code-csharp[Main](first-web-api/sample/TodoApi/Models/TodoRepository.cs)]
+
+Build the app to verify you don't have any compiler errors.
+
+## Register the repository
+
+By defining a repository interface, we can decouple the repository class from the MVC controller that uses it. Instead of instantiating a `TodoRepository` inside the controller we will inject an `ITodoRepository` using the built-in support in ASP.NET Core for [dependency injection](xref:fundamentals/dependency-injection).
+
+This approach makes it easier to unit test your controllers. Unit tests should inject a mock or stub version of `ITodoRepository`. That way, the test narrowly targets the controller logic and not the data access layer.
+
+In order to inject the repository into the controller, we need to register it with the DI container. Open the *Startup.cs* file. 
+
+In the `ConfigureServices` method, add the highlighted code:
+
+[!code-csharp[Main](first-web-api/sample/TodoApi/Startup.cs?name=snippet_AddSingleton&highlight=11)]
+
+## Add a controller
+
+In the *Controllers* folder, create a class named `TodoController`. Add the following (and add closing braces):
+
+[!code-csharp[Main](first-web-api/sample/TodoApi/Controllers/TodoController.cs?name=snippet_todo1)]
+
+This defines an empty controller class. In the next sections, we'll add methods to implement the API.
+
+## Getting to-do items
+
+To get to-do items, add the following methods to the `TodoController` class.
+
+[!code-csharp[Main](first-web-api/sample/TodoApi/Controllers/TodoController.cs?name=snippet_GetAll)]
+
+These methods implement the two GET methods:
+
+* `GET /api/todo`
+
+* `GET /api/todo/{id}`
+
+Here is an example HTTP response for the `GetAll` method:
+
+```
+HTTP/1.1 200 OK
+   Content-Type: application/json; charset=utf-8
+   Server: Microsoft-IIS/10.0
+   Date: Thu, 18 Jun 2015 20:51:10 GMT
+   Content-Length: 82
+
+   [{"Key":"4f67d7c5-a2a9-4aae-b030-16003dd829ae","Name":"Item1","IsComplete":false}]
+   ```
+
+Later in the tutorial I'll show how you can view the HTTP response using [Postman](https://www.getpostman.com/).
+
+### Routing and URL paths
+
+The `[HttpGet]` attribute specifies an HTTP GET method. The URL path for each method is constructed as follows:
+
+* Take the template string in the controller’s route attribute,  `[Route("api/[controller]")]`
+* Replace "[Controller]" with the name of the controller, which is the controller class name minus the "Controller" suffix. For this sample, the controller class name is **Todo**Controller and the root name is "todo". ASP.NET Core [routing](xref:mvc/controllers/routing) is not case sensitive.
+* If the `[HttpGet]` attribute has a template string, append that to the path. This sample doesn't use a template string.
```

Comment:
Right after saying this sample doesn't use a template string there's a snippet with HttpGet and "{id}" which looks like a template string.